### PR TITLE
[Backport-v2.2-branch] Bluetooth: controller: split: correct timing calculation in PKT_US

### LINKED ISSUE #23482: 2M PHY + DLE and timing calculations on an encrypted link are wrong
**Describe the bug**
We are seeing some issue regarding 2M PHY + DLE and timing calculations on an encrypted link.

We use DLE with 251 as maximum and have some internal test that tries to send 244 bytes via GATT with 7 bytes protocol overhead, so that becomes 251 bytes LL PDU Payload. When adding the 4 byte MIC, the LL PDU length becomes 255.

On the AIR this should become one LL packet, instead we are seeing two.

We see that when running 2M that struct lll_conn's max_tx_time is 1060 and when ull_conn_lll_max_tx_octets_get(struct lll_conn *lll) [subsys/bluetooth/controller/ll_sw/ull_conn.c] is called max_tx_octets becomes 1060/8 - 11 = 254 where it should be 255.

The correct max_tx_time should be 1064, with 4us extra for the missing byte.

We believe we have tracked the issue to PKT_US(octets, phy) [subsys/bluetooth/controller/ll_sw/ull_conn_internal.h] where the phy parameter is fixed to 1, when given to PREAMBLE_SIZE(phy) [subsys/bluetooth/controller/ll_sw/pdu.h].

Also we have noticed that in some calls to PKT_US a 0 is given as phy which doesn't make sense, so we guess it should be BIT(0).

**Expected behavior**
On the AIR the 244 bytes GATT operation should be one LL packet.

**Impact**
Throughput degradation.

**Environment**
Zephyr 2.2

### PR DESCRIPTION
**Pull request by kruithofa**
Fixes #23482

A bug in the PKT_US resulted in wrong calculations for the 2M phy since the preamble size was incorrect. This results in reduced throughput since data that should fit in one packet was split in two packets.
Fixes the bug, verified on EBQ.
Also adds some defines for improved readability.

See also PR #23557 which is for the master branch

Signed-off-by: Andries Kruithof <Andries.Kruithof@nordicsemi.no>

### CODE REVIEW

**Review by thoh-ot — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

**Review by wopu-ot — APPROVED** [review]

**Review by dagbja — APPROVED** [review]

**Comment by kruithofa** [comment]
@carlescufi Please advise on what more info is required

**Comment by mtpr-ot** [comment]
@carlescufi, is this still waiting for approval?

**Comment by carlescufi** [comment]
> @carlescufi, is this still waiting for approval?

No, this is in the Ready column of the Backports project. This means that it can be merged at any time, but I will leave it to @jhedberg to decide when.

**Comment by mtpr-ot** [comment]
> > @carlescufi, is this still waiting for approval?
> 
> No, this is in the Ready column of the Backports project. This means that it can be merged at any time, but I will leave it to @jhedberg to decide when.

Fair enough. The sooner the 2.2 backports are merged, the sooner we will start doing serious testing on the branch.

**Review by jhedberg — APPROVED** [review]

**Pull request merged**
